src/data_generators/independent_overlap_data_set.py

```diff
@@ -13,50 +13,48 @@
 # limitations under the License.
 """Generate independent multi-pub data from single-pub data."""
 
+import math
 from typing import Iterable
+from numpy.random import Generator
 from numpy.random import RandomState
 from wfa_cardinality_estimation_evaluation_framework.simulations.set_generator import (
-    IndependentSetGenerator,
-)
+    IndependentSetGenerator,)
 from wfa_planning_evaluation_framework.data_generators.data_set import (
-    DataSet,
-)
+    DataSet,)
 from wfa_planning_evaluation_framework.data_generators.overlap_data_set import (
-    OverlapDataSet,
-)
+    OverlapDataSet,)
 from wfa_planning_evaluation_framework.data_generators.publisher_data import (
-    PublisherData,
-)
+    PublisherData,)
 
 
 class IndependentOverlapDataSet(OverlapDataSet):
-    """Construct a multi-pub DataSet with independent overlap."""
+  """Construct a multi-pub DataSet with independent overlap."""
 
-    def __init__(
-        self,
-        unlabeled_publisher_data_list: Iterable[PublisherData],
-        universe_size: int,
-        random_state: RandomState = None,
-        name: str = "independent",
-    ) -> DataSet:
-        """Constructor for IndependentOverlapDataSet.
+  def __init__(self,
+               unlabeled_publisher_data_list: Iterable[PublisherData],
+               universe_size: int,
+               random_generator: Generator = None,
```

Comment:
Is there a unit test for the case when the random_generator parameter is omitted?  I wonder if it would be better to make this a required parameter?

Reply:
agreed, thank you!